sketch/sketch.ino

```diff
@@ -1,363 +1,486 @@
+    }
+  }
+};
 
-void doPlayerIndication() {
-  Serial.println("Performing Player Indication");
-    // Select starting player.
-  playersTurn = random(0,playerCount);
-  Serial.println("Starting player is Player " + String(playersTurn));
-  highlightPlayerAnimation(playersTurn, 3);
-  Serial.println("Finish Player Indication");
-}
+struct Buzzer {
+  int pin;
+  unsigned long previousToneMillis;
 
-void doGameRunning(int playerCount, byte activePlayers) {
-  Serial.println("Performing GameRunning");
-  // Turn on all LEDs for active players.
-  for (int i = 0; i < playerCount; i++) {
-    digitalWrite(playerLed[i], LOW); // turn off first
-    if (checkIfPlayerActive(i, activePlayers)) {
-      digitalWrite(playerLed[i], HIGH); // turn on active player
-    }
+  void makeNoise() {
+    tone(pin, NOTE_G6, 200);
   }
 
-  readButtonPress(redButtonPin); // will stay here until button is pressed.
+  // TODO: Implement this method so we can play a full melody.
+  // void playMelody() {}
+};
 
-  Serial.println("Finish GameRunning");
-}
 
-void doPlayerSelection() {
-  boolean highlightFlashOn = false;
 
-  // Select a player to highlight
-  doPlayerSelection_selectAPlayer();
+// END  ======  DATA TYPES ====================
 
-  while (true) {
 
-    // Flash currently highlighted player
-    highlightFlashOn = doPlayerSelection_flashHighlightedPlayer(highlightFlashOn);
+// START ====== GLOBAL STUFF ====================
 
-    // MISSING: Show highlighted player triggers and chamber size on some sort of display.
+#define PLAYER_COUNT 4
+byte startingPlayer = -1;
+Player* players = new Player[PLAYER_COUNT];
 
-    if (currentSelectionMode == Knob) {
-      // Read knob direction and change player selection
-      int knobDirection = readKnobDirection();
-      Serial.print("knobDirection: ");
-      Serial.print(knobDirection);
-      Serial.print(" - currentHighlightedPlayer: ");
-      if (knobDirection != 0) {
-        doPlayerSelection_updateHighlightedPlayer(knobDirection);
-      }
-      Serial.println(currentHighlightedPlayer);
-    } else {
-      if (digitalRead(redButtonPin) == LOW) {
-        doPlayerSelection_updateHighlightedPlayer(1); // Select clockwise active player
-        Serial.println("Player selection updated via button");
-      }
-    }
+// LEDs
+const int playerLedPins[] = {2,3,4,5}; // Need to add more pins if more players wanted.
+PlayerLed* playerLeds = new PlayerLed[PLAYER_COUNT];
+const int ledBlinkDuration = 300;
+const int ledIntervalDuration = 500;
 
-    if (digitalRead(greenButtonPin) == LOW) {
-      break;
-    }
-    // Read shoot button press, call logic and then exit loop if pressed
-    // Note: Probably more elegant to change state and handle animation/sounds in new state.
+// Buzzer
+const int buzzerPin = 6;
+struct Buzzer buzzer = {buzzerPin, 0};
 
-    delay(100); //  This delay value important and can affect selection knob accuracy.
-  }
+//  Button
+const int buttonAPin = 7;
+const int buttonBPin = 8;
```

Comment:
All constansts prefer to use constexpr

Reply:
This is a new thing for me, what’s the difference?

Reply:
Sounds pretty cool actually, powerful stuff